### DISCUSSION TITLE
Is there anyway to enable fp64 in pydeck?

### DISCUSSION THREAD

**Discussion by yxia-2022** [visgl/deck.gl#8738] [Q&A]
More context: 
In visualizing point clouds with map overlay, truncation errors can be quite visible. I can confirm that the raw data is in the correct precision, so it's likely that fp64 is not enabled in pydeck automatically.

**Comment by Pessimistress** [comment]
fp64 is automatically enabled for LNGLAT and CARTESIAN coordinate systems. If you are using METER_OFFSETS then it doesn't support fp64. Otherwise depending on how your data is loaded it might be that the loader is passing 32bit data to deck.

**Comment by yxia-2022** [comment]
Thank you so much for the prompt reply! In my code, I find both PointCloudLayer and LineLayer suffer from that truncation problem. In terms of the coordinate, I'm not sure where to set. Here's roughly how my code looks like:
```
    point_cloud_layer = pdk.Layer(
        "PointCloudLayer",
        data=lon_lat_alt_r_g_b_df,
        get_position=["longitude", "latitude", "altitude"],
        get_color=["r", "g", "b"],
        get_normal=[0, 0, 1],
        highlight_color=[255, 255, 0],
        auto_highlight=True,
        pickable=True,
        radiusPixels=2,
    )

    init_view = pdk.ViewState(
        longitude=SOME_MAGIC_LONGITUDE,
        latitude=SOME_MAGIC_LATITUDE,
        controller=True,
        zoom=15,
        max_zoom=50,
        bearing=0,
        pitch=30,
    )
    view = pdk.View(type="MapView", controller=True)
    r = pdk.Deck(
        layers,
        initial_view_state=init_view,
        views=[view],
        map_style="dark",
        tooltip=enable_tooltip,
    )
    r.to_html('pdk.html')
```

**Comment by yxia-2022** [comment]
These are the zigzags that I can see very commonly at roughly meter scale
<img width="468" alt="image" src="https://github.com/visgl/deck.gl/assets/111309004/11d12944-c955-468e-8a14-b67516e43c90">

**Comment by yxia-2022** [comment]
I've tried to ensure that the coordinates are represented using type np.float64 before converting to a dataframe. Maybe the conversion has some problem? But in the generated html, numbers have 14 digits
